packages/@lwc/features/src/babel-plugin/index.js

```diff
@@ -97,18 +92,27 @@ module.exports = function({ types: t }) {
                 }
 
                 const name = propertyPath.node.name;
-                let value = this.featureFlags[name];
+                let value = (this.opts.featureFlags || defaultFeatureFlags)[name];
                 validate(name, value);
 
-                if (!this.opts.prod || value === null) {
+                if (!this.opts.prod) {
                     if (isCompileTimeFlag) {
                         testPath.node.object = t.identifier(RUNTIME_FLAGS_IDENTIFIER);
-                        return;
                     }
+                    return;
                 }
 
                 if (this.opts.prod) {
```

Comment:
Why do we need to differentiate between `PROD` and `DEV` mode in this babel plugin? Both Rollup and Terser do dead code elimination automatically.

Reply:
We talked about this offline but for posterity--`DEV` mode transforms all flags into runtime lookups which cannot be evaluated as dead code and the `PROD` mode transform can potentially be applied on top of that. We could have transformed it back into statically-analyzable code but I went ahead with code modification since it was a simple one-liner.